util/encoding/encoding.go

```diff
@@ -599,3 +600,57 @@ func DecodeIfNull(b []byte) ([]byte, bool) {
 	}
 	return b, false
 }
+
+// EncodeTime encodes a time value, appends it to the supplied buffer,
+// and returns the final buffer. The encoding is guaranteed to be ordered
+// Such that if t1.Before(t2) then after EncodeTime(b1, t1), and
+// EncodeTime(b2, t1), Compare(b1, b2) < 0. The time zone offset is added
+// as a two byte encoding at the end of the time encoding.
+func EncodeTime(b []byte, t time.Time) []byte {
+	// Read the unix time.
+	sec := t.Unix()
+	nsec := t.Nanosecond()
+	if nsec < 0 {
+		// if it were negative we would need to change the way we encode.
+		panic("nanosecond is always non-negative")
+	}
+	b = EncodeVarint(b, sec)
+	b = EncodeVarint(b, int64(nsec))
+
+	// The offset encoding code below is copied from time.MarshalBinary()
```

Comment:
This doesn't seem right, but perhaps I'm missing something. I would have thought that to encode a time you'd want to normalize the offset to UTC. If you don't normalize to UTC, the sorting of times will be off. Normalization to UTC (via `t.UTC()`) also simplifies the encoding. The downside of normalization is that you can't recover the original offset during decoding. Not sure what ramifications that has.